include/internal/print_threadid.h

```diff
@@ -0,0 +1,34 @@
+#include <openssl/crypto.h>
+#include <openssl/e_os2.h>
+
+static ossl_inline size_t thread_to_string(char *buf, size_t len,
+                                           CRYPTO_THREAD_ID tid,
+                                           const char *prefix,
+                                           const char *suffix)
+{
+    size_t n, i, j;
+
+    n = strlen(prefix);
+    if (n + 1 >= len)
+        return 0;
+    memcpy(buf, prefix, n);
+
+    buf += n;
+    n = len - n;
+    for (i = 0; i < sizeof(tid); i++) {
+        if (n < 3)
+            return 0;
+        j = BIO_snprintf(buf, n, "%02X", ((unsigned char *)&tid)[i]);
+        if (j != 2)
+            return 0;
+        buf += j;
+        n -= j;
+    }
+
+    j = strlen(suffix) + 1;
+    if (j >= n)
+        return 0;
+    n -= j;
+    memcpy(buf, suffix, j);
+    return len - n;
+}
```

Comment:
Another hexdump...  and not even generic, but specialised for thread IDs.

I'm sorry, this is not ok.  The best would be to use an already available hex dumper (we already have a few BIO hex printers in `crypto/bio/b_dump.c`, and there's the generic `OPENSSL_buf2hexstr`).  However, if none of them pleases you, *at the very least* make something generic, not something that's specialised for just thread IDs!

Reply:
I'd forgotten about buf2hexstr.  The BIO based ones didn't seem appropriate.

I'll change.

Reply:
Using `OPENSSL_buf2hexstr` greatly simplified the code thanks.  Update pushed.

Reply:
An idea I have in the back of my head is to extend the printf formating we currently understand, to allow direct printing of, say, BIGNUMs, and why not hex dumping of any given memory...

Reply:
I've thought of BIGNUM printf formatting too.  It will break compiler printf checking :(

Reply:
We've broken it already for some compiler versions, 'cause we were a step ahead...

Between that and limiting ourselves stoopidly...  I dunno.